src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -102,12 +103,27 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
-	@SuppressWarnings( "unchecked" )
+	public void save( final OutputStream outputStream, final T element ) throws IOException {
+		final Output output = new Output( new LZ4FrameOutputStream( outputStream ) );
+		output.writeString( version );
+		kryo.writeClassAndObject( output, element );
+		output.close();
+	}
+
 	@Override
 	public T load( final URI identifier ) throws IOException {
 		final Path path = Paths.get( identifier );
+		try {
+			return load( newInputStream( path ), identifier );
+		} catch ( final IncompatibleReportVersionException | NoSuchFileException e ) {
+			throw e;
+		}
+	}
+
+	@SuppressWarnings( "unchecked" )
+	public T load( final InputStream in, final URI identifier ) throws IOException {
```

Comment:
Do you have `URI` with your stream? If not, I would try to remove the identifier (as it is only used in the exception). I would generally try to keep the methods the same:
```
T load( final URI identifier ) throws IOException;
T load( final InputStream identifier ) throws IOException;
```

Reply:
Yeah, I tried that ... but I think this is really helpful information in case of errors. So how could you remove that then?